### PR TITLE
8768 :: Resolve Addrange issue after sorting.

### LINKED ISSUE #8768: TreeNodeCollection.AddRange works incorrectly for non-empty collection of the sorted TreeView
### .NET version

6.0

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

When the tree nodes are appended to the non-empty `TreeNodeCollection` owned by the sorted `TreeView` with already created underlying window, the resulting order of items in the actual window does not match the one in the collection (which is correct).

The cause is how method `TreeNode.PrevNode` works: when called from within `TreeNodeCollection.AddRange` of non-empty collection, it always uses fixed index as the position of the previous node, assuming the normal backward insertion of the nodes in the unsorted tree. Yet in case of the sorted `TreeView`, insertions are made at various positions according to a sort order.

There obviously should be a check for the sorted state of the tree in `TreeNode.PrevNode` to avoid this.

### Steps to reproduce

```
namespace TreeViewTest
{
  internal static class Program
  {
    [STAThread]
    static void Main()
    {
      ApplicationConfiguration.Initialize();

      var treeView = new TreeView();
      treeView.Nodes.Add(new TreeNode("2"));
      treeView.Sort();

      var form = new Form();
      form.Controls.Add(treeView);
      form.Load += (sender, e) => treeView.Nodes.AddRange(new TreeNode[] {
        new TreeNode("3"),
        new TreeNode("5"),
        new TreeNode("1"),
        new TreeNode("4")
      });
      Application.Run(form);
    }
  }
}
```

This results in:
Clipboard01

### PR DESCRIPTION
**Pull request by RutulPatel8**
Fixes #8768 

## Proposed changes
if tree is already sorted then we don't want to set fixed Index value. due to these reason It is misbehave. 

## Customer Impact

- No

## Regression? 

- Yes

## Risk

- No

### Before

When the user adds a value to the treenode list and sorts it using treeview.sort(), it works fine. Subsequently, when the user adds individual trees through the add event, it functions as expected and sorts the entire tree automatically. However, if the user adds multiple values through AddRange, it does not work as expected.

### After
I've fixed the sorting issue we discussed earlier, and now everything is working smoothly. When users add values individually or use treeview.sort(), the sorting works fine. Even when they add multiple values at once using AddRange, the sorting behaves as expected.

## Test methodology 

- Testcase added

## Test environment(s)

- tested with .net core 8.0

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11300)

### CODE REVIEW

**Comment by Tanya-Solyanik** [comment]
@RutulPatel8  - please investigate test failures and take a look at another PR in TreeNodesCollection area that might be addressing this regression as well
https://github.com/dotnet/winforms/pull/11253.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeCollectionTests.cs [hunk 1]
```diff
@@ -214,6 +214,39 @@ public void TreeNodeCollection_Find_InvokeNoSuchKey_ReturnsEmpty(string key)
...
+        };
+        TreeNode child3 = new()
+        {
+            Name = "7"
+        };
+
+        treeView.Nodes.Add(child1);
+        treeView.Nodes.Add(child2);
+        treeView.Nodes.Add(child3);
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeCollectionTests.cs:238]: After the nodes are added you can verify that they are not sorted. And after you sort them, you can verify that they are sorted

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeCollectionTests.cs:238]: As far as I understand, this issue repros only when AddRange is called from the Load event handler, i.e. when AddRange is called after the native TreeView had been initialized, please include that in this test.

**Review by Tanya-Solyanik — COMMENTED** [review]
The approach to this fix seems correct, but this fix causes failures in TreeView tests, please investigate them. Also this kind on a change requires an opt-in switch, like the one @elachlan added here - https://github.com/dotnet/winforms/pull/10493/files

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeNodeCollection.cs [hunk 2]
```diff
@@ -219,7 +219,11 @@ public virtual void AddRange(params TreeNode[] nodes)
             tv.BeginUpdate();
         }
 
-        _owner.Nodes.FixedIndex = _owner._childNodes.Count;
+        if (tv is not null && !tv.Sorted)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeNodeCollection.cs:222]:
```suggestion
        if (tv is null || !tv.Sorted)
```

This is the reason of the test failures

Comment (RutulPatel8) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeNodeCollection.cs:222]: Let me correct it.

**Pull request closed without merging**